tlslite/tlsconnection.py

```diff
@@ -1730,6 +1758,15 @@ def _handshakeServerAsyncHelper(self, verifierDB,
             extensions.append(ECPointFormatsExtension().create(
                 [ECPointFormat.uncompressed]))
 
+        # if client sent Heartbeat extension
+        if clientHello.getExtension(ExtensionType.heartbeat):
```

Comment:
why the value of the client sent extension is ignored here?

Reply:
Value of the client sent extension is checked in '_serverGetClientHello' function (at line 2615) and after check all attributes (if we can send/receive etc.) are set. 
And here, in mentioned line, is just being checked, if client wants Heartbeat. If he wants it, then we send it too.